### PR TITLE
FALCON-2073 Handle with NULL corner case

### PR DESCRIPTION
**Pull request by yzheng-hortonworks**
Changes:
1. EntityGraph::getDependents should return empty list instead of NULL value if there is no dependent entities. Affected method include OozieWorkflowEngine::updateDependant and AbstractEntityManager::getDependencies. Also removed unnecessary existing NULL checks after the change.
2. In LogProvider::populateActionLogUrls, handle with the NULL case where there is no file status under the specified path.
3. FalconClient::getDependency should return empty list instead of NULL value if there is no dependent entities. Affected method include FalconEntityCLI::entityCommand.

Extra minor changes in this patch:
4. A regular expression misusage in LogProvider::getActionName. To match special character ".", should use "[.]" instead of "." which will try to match any character.
5. Fix a rat check error in Falcon CLI due to the introduction of Spring shell commands.

### CODE REVIEW

File: client/src/main/java/org/apache/falcon/client/FalconClient.java [hunk 1]
```diff
@@ -471,7 +471,7 @@ public EntityList getDependency(String entityType, String entityName, String doA
 
         EntityList result = clientResponse.getEntity(EntityList.class);
         if (result == null || result.getElements() == null) {
```
Comment (bvellanki) [inline client/src/main/java/org/apache/falcon/client/FalconClient.java]: You dont need the " result.getElements() == null " condition

**Comment by bvellanki** [comment]
Apart from minor nit, +1

File: client/src/main/java/org/apache/falcon/client/FalconClient.java (hunk 1 above)
Comment (yzheng-hortonworks) [inline client/src/main/java/org/apache/falcon/client/FalconClient.java]: Removed. Thanks @bvellanki for review.

**Comment by peeyushb** [comment]
Patch looks good. +1 .

**Comment by vrangan** [comment]
+1  Thanks @yzheng-hortonworks   please mention that these were from the coverity scan to give due credit for finding the issues

**Comment by yzheng-hortonworks** [comment]
@vrangan Thanks for review. It is mentioned in the JIRA already :)

**Comment by peeyushb** [comment]
If no further review comments, I will commit this patch soon.

**Pull request closed without merging**